File: JavaScript/7js 3/solve_fixstring.js
/*
|--------------------------------------------------------------------------
| Fix string case
|--------------------------------------------------------------------------
|
*/

function solve(s) {
   return [...s].map((i) => i == i.toLowerCase()).filter((i) => i == true).length < s.length / 2
      ? s.toUpperCase()
      : s.toLowerCase();
}


function solve(s) {
   return s.replace(/[A-Z]/g, "").length < s.length / 2 ? s.toUpperCase() : s.toLowerCase();
}


function solve(s) {
   return [...s].reduce((t, c) => t + /[a-z]/.test(c) * 2, 0) < s.length
      ? s.toUpperCase()
      : s.toLowerCase();
}


console.log(solve("code"), "code");
console.log(solve("CODe"), "CODE");
console.log(solve("COde"), "code");
console.log(solve("Code"), "code");

File: Python/7py 3/solve_fixstring.py
"""
|--------------------------------------------------------------------------
| Fix string case
|--------------------------------------------------------------------------
|
"""

def solve(s):
	return s.upper() if len(list(filter(lambda i: i.islower(), s))) < len(s)/2 else s.lower()


def solve(s):
	return (s.lower, s.upper)[sum(map(str.isupper, s)) > len(s) / 2]()


def solve(s):
	return s.upper() if sum(i.isupper() for i in s) > len(s) / 2 else s.lower()
	

print(solve("code"),"code")
print(solve("CODe"),"CODE")
print(solve("COde"),"code")
print(solve("Code"),"code")